server/datastore/mysql/migrations/tables/20220915165115_AppleMDMTables_nanomdm.sql

```diff
@@ -79,6 +79,7 @@ CREATE TABLE nano_users (
     CHECK (user_authenticate_digest IS NULL OR user_authenticate_digest != '')
 );
 
+ALTER TABLE nano_users ADD CONSTRAINT idx_unique_id UNIQUE (id);
```

Comment:
Maybe there's a reason the primary key is id+device_id? (to support multiple user enrollments on the same device?)

Reply:
If the `nano_users.id` is non-unique, then this foreign key constraint in `nano_enrollments` is bad:

```
    FOREIGN KEY (user_id)
        REFERENCES nano_users (id)
        ON DELETE CASCADE ON UPDATE CASCADE,
```

It maybe should be
```
    FOREIGN KEY (user_id,device_id)
        REFERENCES nano_users (id,device_id)
        ON DELETE CASCADE ON UPDATE CASCADE,
```

Reply:
I'm guessing we can worry about this when we decide to support [user enrollments](https://support.apple.com/guide/deployment/user-enrollment-and-mdm-dep23db2037d/web).

Reply:
I checked Dogfood, and `nano_users.id` is always unique there. `nano_users.device_id` is not unique.

Reply:
After looking at the code, I believe this PR's approach is correct -- `nano_users.id` is unique.

In the code, we always update `nano_enrollments` after updating `nano_users`. `nano_enrollments` already has a unique constraint on `user_id`, and `nano_enrollments.id` is always either `nano_enrollments.user_id` or `nano_enrollments.device_id`. So, in effect we are already enforcing the above-mentioned unique constraint in the code.